### ISSUE TITLE
How do you actually use this?

### ISSUE THREAD

**Issue by sergiotapia** [jigish/slate#381]
I've tried reading the documentation but it's very overwhelming. I really don't care for elaborate configs and whatnot. I just want to be able to size windows 50%50% side by side using keystrokes similar to Windows 8.

Can anyone give a hand? This is really user unfriendly. :(

**Comment by JolinM** [comment]
In my opinion, what make the beauty of slate is it's complexity. If you want something simpler.
[ShiftIt](https://github.com/fikovnik/ShiftIt) seams to be a popular free alternative.
Otherwise, go have a look at these lists, there's a great diversity there!
http://apple.stackexchange.com/questions/9426/snap-feature-for-mac
http://apple.stackexchange.com/questions/9659/what-window-management-options-exist-for-os-x

**Comment by jaseemabid** [comment]
The README clearly lacks a 'getting started with slate'. I am very familiar with tiling window managers and needed an alternative to the amazing i3 when I moved to mac from linux for a while. 

I installed it, fixed system preferences etc. Now tell me how I can quickly do something rather than make me read a exceptionally long README. Who reads an entire manual before trying out something? Its absurd.

A video tutorial will be really helpful and I'm surprised that there is none on YouTube.

**Comment by sergiotapia** [comment]
I agree, believe me if I had the time I would read up on it, use it, and write a get started guide but as it is I don't have the time to invest at the moment. I went with Spectacle, which does what I want intuitively. http://spectacleapp.com/

**Comment by adam-jensen** [comment]
Stanford computer science student, no idea how to use this thing.

**Comment by jaseemabid** [comment]
@adam-jensen  LOL :)

The default mac window manager is unbelievably fucked up compared to the amazing options you have on linux like awesome, i3, xmonad, dwm and rat poison. Clearly lacks a decent tiling window manager. Time to move on :)

**Comment by JolinM** [comment]
These two examples are quite interesting to understand how the system works. Take a look at their config files. It's a good way to get started!

[using slate](http://thume.ca/howto/2012/11/19/using-slate/)
[simple slate configuration](http://larryhynes.net/2013/11/simple-slate-configuration.html)

**Comment by jaseemabid** [comment]
@JolinM You are completely missing the point. Please have a look at i3 doc[1]. It tells you how you can do _something_ with it without reading the entire manual. I want to see it in action before I commit the time, and so will be the mentality of majority of the users. 

And _I_ belong to the category of programmers who tweak tools a lot!  I have tracked my dotfiles for over 2 years[2], have read the 1000+ page emacs manual cover to cover and have a  500+ line init.el[3]. I do it because I know the tool is good and worth tweaking and spending time on. I don't know that about slate yet.  

So tldr; Put the default shortcut to make a window occupy 50% width and fundamentals people expect out of a window manager on top of the README. Then let users follow though the doc and do more and more things, tweaking as they read on.
1. http://i3wm.org/docs/userguide.html
2. https://github.com/jaseemabid/dotfiles
3. https://github.com/jaseemabid/emacs.d/blob/master/init.el

**Comment by sergiotapia** [comment]
:+1:

**Comment by JolinM** [comment]
Got it.
This would indeed be a great addition.

**Comment by jaseemabid** [comment]
This is what's coming to my mind now. 
![Learning curve](http://i.stack.imgur.com/7Cu9Z.jpg)

This is how I would describe slate

```
     +
     |
     | +----+     <=  [Finding the right keyboard shortcut]
     |      |
     |      |
     |      |
     |      |
     |      |
     |      |
     |      |
     |      |
     |      |
     |      |
     |      |
     |      ------------------------------>
     -------------------------------------------->
```

**Comment by harizvi** [comment]
You mean something like here: 
https://github.com/jigish/slate/wiki/JavaScript-Configs 

It tells you how you can do something pretty crazy with slate, without making you read the whole manual first.

**Comment by jaseemabid** [comment]
@harizvi No! That tutorial tells me I can copy some js to a config file and make `Ctrl+1` do something  almost useless. 

Ok, let me rephrase it. 
1. I don't want to configure slate before I use it for a while and get the hang of it => Manuals for an app before you do something with it is utter useless. 
2. What can it do _OUT OF THE BOX_ ?
3. Is there one key binding like 'Ctrl+Command+j+k` to make it do _SOMETHING_ even completely useless like making a window blink so that the user can verify the app is installed right and is actually working?

**Comment by sergiotapia** [comment]
KEY KEY KEY!

> What can it do OUT OF THE BOX ?

This needs to be put front and center in the README.

**Comment by ranman** [comment]
"how I can quickly do something rather than make me read a exceptionally long README"
...
Time to complain about this in an issue but not enough time to read the README or even better since it's an open source project... contribute to the README? _smh_

**Comment by jaseemabid** [comment]
@ranman Just like @sergiotapia said, I'd love to write a quick intro _after_ I get this working for me nicely.

The general idea about README is to give a quick overview, and you can move all configuration to a wiki or another file to make life easy for users. Please assume you know **nothing** about slate or a while and read it again, and tell me its any good for a first timer.

**Comment by ranman** [comment]
Just gave it a once over with your advice. I see your point :+1:, my apologies!

I think I've just become an elitist programming snob after trying to parse all of these opaque and ancient linux man pages. Compared to those this README is golden.

I'll get to work on a PR that has a quick-start in the README. Although... it looks like this project hasn't been actively developed in quite a while :-1: .

**Comment by jaseemabid** [comment]
@ranman No worries man, I was a bit pissed off too last night when I wrote that :/ I got here from a window manager comparison[2] on slant.co. I'll be spending sometime on this later today. Will see how the docs can be improved.
1. http://www.slant.co/topics/526/~what-is-the-best-window-manager-for-mac

**Comment by sergiotapia** [comment]
Thanks a bunch for donating your time @ranman , we all appreciate it! :)

**Comment by ky1ejs** [comment]
I'm a 1st year Software Engineering student from London and have been slowly getting to grips with Slate (quite comfortable with it now) over the past few weeks. I've been playing around with the idea of writing a blog post for new developers and perhaps even a new project to create a UI for slate to make it easy to configure. @jaseemabid  since you're so keen to see something like this, maybe you'd like to help? Having the guide written by newbies as they go along would make it so easy to pick up for other newbies...

I've been using it in conjunction with KeyRemap4MacBook (https://pqrs.org/macosx/keyremap4macbook/) and PCKeyboardHack (https://pqrs.org/macosx/keyremap4macbook/pckeyboardhack.html.en) which I think are a know brainer to install if you're using slate.

**Comment by jaseemabid** [comment]
@kylejm Mac is not my primary dev platform, and I'm soon switching back to linux. I wont be using slate much

**Comment by ProLoser** [comment]
If you guys don't feel like figuring it out yourself, you should use mine: https://github.com/ProLoser/reslate

copy this folder to `~/.reslate` and copy `slate.example.js` to ~/.slate.js`

I am still working on features, but for now you can do <kbd>ctrl</kbd>+<kbd>option</kbd>+<kbd>command</kbd>+<kbd>arrows</kbd> (or <kbd>J</kbd><kbd>K</kbd><kbd>L</kbd><kbd>I</kbd>) to move windows around and across monitors.

Right now I'm discussing with @lunixbochs a better key configuration than what we're doing now and I'm trying to implement auto-snapshotting, meaning when you plug/unplug monitors your windows all revert to their last configuration that was used with those monitors.

**Comment by RyanBalfanz** [comment]
If it help: https://github.com/RyanBalfanz/dotfiles/blob/master/slate.js

**Comment by tphalp** [comment]
@ProLoser and @RyanBalfanz - Thank you both.

**Comment by ProLoser** [comment]
Mine needs a fair bit of work lately lol.
